### PR TITLE
execute.c: support pick(last)

### PR DESCRIPTION
**Pull request by pkoppstein**
Support [0] | setpath([-1]; _)
Add tests for pick/1

Supersedes https://github.com/jqlang/jq/pull/2717

### CODE REVIEW

File: src/execute.c [hunk 1]
```diff
@@ -687,6 +687,14 @@ jv jq_next(jq_state *jq) {
...
         goto do_backtrack;
       }
+      // $array | .[-1]
+      if (jv_get_kind(k) == JV_KIND_NUMBER && jv_get_kind(t) == JV_KIND_ARRAY) {
+        int idx = jv_number_value(k);
+        if (idx < 0) {
+          jv_free(k);
+          k = jv_number(jv_array_length(jv_copy(t)) + idx);
+        }
+      }
```
Comment (emanuele6) [inline src/execute.c:697]: This patch should be authored by @nicowilliams

**Comment by nicowilliams** [comment]
@pkoppstein please do nothing.  I'll push my version of this with attribution to you.

**Comment by nicowilliams** [comment]
.

**Comment by nicowilliams** [comment]
Ok, @pkoppstein I've force-pushed to this PR.  Please do not push to it again.  Just tell me if it now has the contents that you wanted.  If it does then I'll merge it after all the tests pass.

**Comment by nicowilliams** [comment]
@pkoppstein I've added the following wiki page for you: https://github.com/jqlang/jq/wiki/Guide-for-Contributors

File: src/execute.c (hunk 1 above)
Comment (pkoppstein) [reply src/execute.c:697]: @emanuele6 - It was.

**Comment by pkoppstein** [comment]
@nicowilliams wrote:
> Just tell me if it now has the contents that you wanted. If it does then I'll merge it after all the tests pass.

It's missing one minor comment, but that's OK.  I see that "All checks have passed" so please merge it before things get complicated again. Thank you.

**Comment by nicowilliams** [comment]
> It's missing one minor comment, but that's OK. I see that "All checks have passed" so please merge it before things get complicated again. Thank you.

OK, we're in the middle of trying to create a signed `jq-1.7rc1`.  We'll merge this after that.

**Review by emanuele6 — APPROVED** [review]

**Pull request merged**